libs/xmlforms/cleaner.js

```diff
@@ -0,0 +1,60 @@
+var _ = require('underscore'),
+    rsvp = require('rsvp'),
+    log = require('loglevel'),
+    format = require('pg-format');
+
+var getTablesForDocs = function(db, docIds) {
+  return db.query(format('SELECT uuid, formname, formversion FROM form_metadata WHERE uuid in (%L)', docIds))
+  .then(function(results) {
+    log.info('Found ' + results.length + ' documents to be removed before regenerating');
+
+    var grouped =_.groupBy(results, function(row) {
+      return 'formview_' + row.formname + '_' + row.formversion;
+    });
+
+    _.chain(grouped)
+      .keys()
+      .each(function(key) {
+        grouped[key] = _.pluck(grouped[key], 'uuid');
+      });
```

Comment:
It'd be nicer to just use native JS here, eg: `Object.keys(grouped).forEach(function(key) {...`